Add source list and ability to save results to csv

bradygm opened the pull request:
### Source List
You can specify which ROS nodes you would like to monitor by setting the `source_list` argument when launching. The default is an empty list which then monitors all ROS nodes.

Setting `source_list` in the command line:
```
roslaunch cpu_monitor cpu_monitor.launch source_list:="[<node_name_1>, <node_name_2>]"
```

Or you can set the default values in the included launch file `cpu_monitor.launch`

### Saving to CSV
You can save the results to a CSV by setting the `save_to_csv` parameter to true. The results will be saved to the relative path specified in `csv_file` parameter. By default the file is saved in the folder under a file names `cpu_monitor.csv`. If `save_to_csv` is set to true, the node waits for 5 seconds for other nodes to start up before saving the list of nodes it will be monitoring. This is so that the header of the CSV stays constant throughout the tests. If new nodes pop up, they will not be recorded, as this would change the header of the CSV.

bradygm committed:
Add source list to only modify subset of nodes

bradygm committed:
Allow for saving results to csv

alspitz commented on monitor.py:
```diff
@@ -58,45 +73,74 @@ def alive(self):
+      for node_name, node in list(node_map.items()):
+        if node.alive():
+          new_csv_line += node.publish()
+        else:
```
Would it be possible to have this keep track of nodes that die? If a node dies while running, the CSV would have the stats under the wrong header for nodes that come after the dead node, right? May be good to have a CSVWriter class that knows what nodes it's tracking, and then accepts a dict : node -> stats, and composes the line with the stats in the appropriate slot. Can also help with only having one loop over the items in node_map. At the end can have a small if save_to_csv block that passes the collected stats (returned from publish() maybe) into the CSVWriter object.

alspitz commented:
Thanks for these changes! Made some suggestions in a comment above. Also, would it be possible to have the node continue to track new nodes without affecting the CSV behavior? So, would still track new nodes to publish stats for even if save_to_csv is true, it's just that the data of the new nodes will not end up in the CSV. May be good to enforce that publish behavior remains the same regardless of save_to_csv. Let me know what you think!

bradygm committed:
Created CSVWriter class.

- Publishing behaviro remains the same regardless of save_to_csv.

- The source list parameter only applies to saving to csv.

- CSV saver can handle if nodes die, and it will not mess up the
alignment of the columns.

bradygm commented:
@alspitz, I incorporated a lot of your suggestions and fixed the issues with dropping and adding nodes. 
* CSVWriter class 
* Publishing behavior remains the same regardless of save_to_csv.
* The source list parameter only applies to saving to csv.
* CSV saver can handle if nodes die, and it will not mess up the alignment of the csv columns.

The only thing I did not do is minimize everything to one loop. However, it will only loop over the selected topics to be saved and it just has to lookup the saved info on each node and doesn't loop over the entire node_map again. 

Let me know what you think.

alspitz commented:
@bradygm Looks great, thanks!
